stream/src/main/java/org/apache/kafka/clients/processor/internals/ProcessorContextImpl.java

```diff
@@ -0,0 +1,267 @@
+                if (!topics.contains(topic))
+                    return false;
+            }
+        }
+
+        return true;
+    }
+
+    @Override
+    public int id() {
+        return id;
+    }
+
+    @Override
+    public Serializer<?> keySerializer() {
+        return processorProperties.keySerializer();
+    }
+
+    @Override
+    public Serializer<?> valueSerializer() {
+        return processorProperties.valueSerializer();
+    }
+
+    @Override
+    public Deserializer<?> keyDeserializer() {
+        return processorProperties.keyDeserializer();
+    }
+
+    @Override
+    public Deserializer<?> valueDeserializer() {
+        return processorProperties.valueDeserializer();
+    }
+
+    @Override
+    public RecordCollector recordCollector() {
+        return collector;
+    }
+
+    @Override
+    public File stateDir() {
+        return stateMgr.baseDir();
+    }
+
+    @Override
+    public Metrics metrics() {
+        return metrics;
+    }
+
+    @Override
+    public void register(StateStore store, RestoreFunc restoreFunc) {
+        if (initialized)
+            throw new KafkaException("Can only create state stores during initialization.");
+
+        stateMgr.register(store, restoreFunc);
+    }
+
+    @Override
+    public void flush() {
+        stateMgr.flush();
+    }
+
+    public String topic() {
+        if (streamGroup.record() == null)
+            throw new IllegalStateException("this should not happen as topic() should only be called while a record is processed");
+
+        return streamGroup.record().topic();
+    }
+
+    @Override
+    public int partition() {
+        if (streamGroup.record() == null)
+            throw new IllegalStateException("this should not happen as partition() should only be called while a record is processed");
+
+        return streamGroup.record().partition();
+    }
+
+    @Override
+    public long offset() {
+        if (this.streamGroup.record() == null)
+            throw new IllegalStateException("this should not happen as offset() should only be called while a record is processed");
+
+        return this.streamGroup.record().offset();
+    }
+
+    @Override
+    public long timestamp() {
+        if (streamGroup.record() == null)
+            throw new IllegalStateException("this should not happen as timestamp() should only be called while a record is processed");
+
+        return streamGroup.record().timestamp;
+    }
+
+    @Override
+    public void send(String topic, Object key, Object value) {
+        collector.send(new ProducerRecord<>(topic, key, value));
+    }
+
+    @Override
+    public void send(String topic, Object key, Object value, Serializer<Object> keySerializer, Serializer<Object> valSerializer) {
+        if (keySerializer == null || valSerializer == null)
+            throw new IllegalStateException("key and value serializers must be specified");
+
+        collector.send(new ProducerRecord<>(topic, key, value), keySerializer, valSerializer);
```

Comment:
I think we can wrap the producer / consumer configs in the streaming / processor congis as you mentioned.